Common/Data/Custom/CBOE/CBOE.cs

```diff
@@ -0,0 +1,123 @@
+using System.IO;
+using System.Linq;
+
+namespace QuantConnect.Data.Custom.CBOE
+{
+    public class CBOE : TradeBar
+    {
+        /// <summary>
+        /// Gets the source location of the CBOE file
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="date"></param>
+        /// <param name="isLiveMode"></param>
+        /// <returns></returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            if (isLiveMode)
+            {
+                return new SubscriptionDataSource($"http://cache.quantconnect.com/alternative/cboe/{config.Symbol.Value.ToLowerInvariant()}.csv", SubscriptionTransportMedium.RemoteFile);
+            }
+
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "alternative",
+                    "cboe",
+                    $"{config.Symbol.Value.ToLowerInvariant()}.csv"
+                ),
+                SubscriptionTransportMedium.LocalFile
+            );
+        }
+
+        /// <summary>
+        /// Reads the data from the source and creates a BaseData instance
+        /// </summary>
+        /// <param name="config">Configuration</param>
+        /// <param name="line">Line of data</param>
+        /// <param name="date">Date we're requesting data for</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>New BaseData instance to be used in the algorithm</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            if (line.StartsWith("Cboe") || line.StartsWith("Date") || line.StartsWith(","))
+            {
+                return null;
+            }
+
+            var csv = line.Split(',')
+                .Select(x => x.Trim())
+                .ToList();
+
+            decimal open;
+            decimal high;
+            decimal low;
+            decimal close;
+
+            if (!decimal.TryParse(csv[1], NumberStyles.Any, CultureInfo.InvariantCulture, out open))
+            {
+                return null;
+            }
+            if (!decimal.TryParse(csv[2], NumberStyles.Any, CultureInfo.InvariantCulture, out high))
+            {
+                return null;
+            }
+            if (!decimal.TryParse(csv[3], NumberStyles.Any, CultureInfo.InvariantCulture, out low))
+            {
+                return null;
+            }
+            if (!decimal.TryParse(csv[4], NumberStyles.Any, CultureInfo.InvariantCulture, out close))
+            {
+                return null;
+            }
+
+            return new CBOE
+            {
+                // Add a day delay to the data so that we LEAN doesn't assume that we get the data at 00:00 the day of
+                Time = DateTime.Parse(csv[0], CultureInfo.InvariantCulture).AddDays(1),
```

Comment:
Please use the helper methods added to cleanly deal with culture settings. For parsing, see the [`Parse`](https://github.com/QuantConnect/Lean/blob/master/Common/Parse.cs) class.